diff --git a/node_modules/signature_pad/dist/signature_pad.js b/node_modules/signature_pad/dist/signature_pad.js
index ca7b3b2..16e386b 100644
--- a/node_modules/signature_pad/dist/signature_pad.js
+++ b/node_modules/signature_pad/dist/signature_pad.js
@@ -283,12 +283,14 @@ SignaturePad.prototype._strokeUpdate = function (event) {
       this._drawCurve(curve, widths.start, widths.end);
     }
 
-    this._data[this._data.length - 1].push({
-      x: point.x,
-      y: point.y,
-      time: point.time,
-      color: this.penColor
-    });
+    if (this._data && this._data[this._data.length - 1]) {
+      this._data[this._data.length - 1].push({
+        x: point.x,
+        y: point.y,
+        time: point.time,
+        color: this.penColor
+      });
+    }
   }
 };
 
@@ -300,7 +302,7 @@ SignaturePad.prototype._strokeEnd = function (event) {
     this._drawDot(point);
   }
 
-  if (point) {
+  if (point && this._data != null) {
     var lastPointGroup = this._data[this._data.length - 1];
     var lastPoint = lastPointGroup[lastPointGroup.length - 1]; // plain object
 
diff --git a/node_modules/signature_pad/dist/signature_pad.mjs b/node_modules/signature_pad/dist/signature_pad.mjs
index ce5f5a4..85984b0 100644
--- a/node_modules/signature_pad/dist/signature_pad.mjs
+++ b/node_modules/signature_pad/dist/signature_pad.mjs
@@ -277,12 +277,14 @@ SignaturePad.prototype._strokeUpdate = function (event) {
       this._drawCurve(curve, widths.start, widths.end);
     }
 
-    this._data[this._data.length - 1].push({
-      x: point.x,
-      y: point.y,
-      time: point.time,
-      color: this.penColor
-    });
+    if (this._data && this._data[this._data.length - 1]) {
+      this._data[this._data.length - 1].push({
+        x: point.x,
+        y: point.y,
+        time: point.time,
+        color: this.penColor
+      });
+    }
   }
 };
 
@@ -294,7 +296,7 @@ SignaturePad.prototype._strokeEnd = function (event) {
     this._drawDot(point);
   }
 
-  if (point) {
+  if (point && this._data != null) {
     var lastPointGroup = this._data[this._data.length - 1];
     var lastPoint = lastPointGroup[lastPointGroup.length - 1]; // plain object
 
